### PR TITLE
react 15.5 support, refactor tests to check multiple react versions

### PR DESCRIPTION
**Pull request by modosc**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,10 +5,14 @@
   "main": "skin-deep.js",
   "typings": "skin-deep.d.ts",
   "scripts": {
-    "test": "mocha",
     "lint": "eslint . --max-warnings 0",
     "coverage": "istanbul cover _mocha --",
-    "doc": "doctoc README.md"
+    "doc": "doctoc README.md",
+    "test": "npm run react:14 && mocha && npm run react:15 && mocha && npm run react:155 && mocha",
+    "react:clean": "rimraf node_modules/react node_modules/react-dom node_modules/react-addons-test-utils react-test-renderer react-addons-create-fragment @types/react create-react-class prop-types",
```
Comment (glenjamin) [inline package.json]: The test suite already runs on multiple react versions, using the environment matrix feature of travis CI

See `.travis.yml` and `install-relevant-react.sh` for details.

File: package.json [hunk 2]
```diff
@@ -37,14 +41,11 @@
   },
   "devDependencies": {
     "@glenjamin/eslint-config": "^2.1.0",
-    "@types/react": "^0.14.57",
```
Comment (glenjamin) [inline package.json]: Could you leave the dev dependencies in for the latest react version?
The script which overrides these and installs the right ones will just ignore them anyway.

File: package.json (hunk 1 above)
Comment (modosc) [inline package.json]: i totally missed that, done.

File: package.json (hunk 2 above)
Comment (modosc) [inline package.json]: done.

File: install-relevant-react.sh [hunk 3]
```diff
@@ -4,8 +4,24 @@ REACT=${REACT:-0.14}
 
 echo "installing React $REACT"
 
+packages=$(cat <<-END
```
Comment (glenjamin) [inline install-relevant-react.sh]: Might be simpler just to do two different calls to `npm install`?

File: package.json [hunk 4]
```diff
@@ -5,10 +5,11 @@
   "main": "skin-deep.js",
   "typings": "skin-deep.d.ts",
   "scripts": {
-    "test": "mocha",
     "lint": "eslint . --max-warnings 0",
     "coverage": "istanbul cover _mocha --",
-    "doc": "doctoc README.md"
+    "doc": "doctoc README.md",
+    "test": "mocha",
+    "test:all": "REACT=0.14 sh ./install-relevant-react.sh && npm run test && REACT=15.4 sh ./install-relevant-react.sh && npm run test && REACT=15.5 sh ./install-relevant-react.sh && npm run test"
```
Comment (glenjamin) [inline package.json]: Do you think it's worth pulling this out into a script which reads the list of versions from `.travis.yml` ?

I'm not sure, might be neater going forwards if React keeps playing around with these APIs

File: install-relevant-react.sh (hunk 3 above)
Comment (modosc) [inline install-relevant-react.sh]: done

File: package.json (hunk 4 above)
Comment (modosc) [inline package.json]: done

**Pull request merged**